### DISCUSSION TITLE
[Question] How to make franka_alt_finger.usd or custom USDs instanceable in IsaacLab

### DISCUSSION THREAD

**Discussion by right-chan** [isaac-sim/IsaacLab#1081] [Q&A]
Hi, I want to make other USDs, such as franka_alt_finger or custom robots, instanceable in IsaacLab. 

I noticed a structural difference between franka.usd and franka_instanceable.usd in the following path:
**omniverse://localhost/NVIDIA/Assets/Isaac/4.2/Isaac/Robots/Franka/**
- `franka.usd` structure:
```
panda (defaultPrim) (Xform)
    panda_link0 (Xform)
        panda_joint1 (PhysicsRevoluteJoint)
        geometry (Xform)
            panda_link0 (Mesh)
                subset (GeomSubset)
                subset_1 (GeomSubset)
                ...
            Looks (Scope)
                EmissiveBlue (Material)
                    Shader (Shader)
                PlasticGray (Material)
                    Shader (Shader)
                ...
```
- `franka_instanceable.usd` structure:
```
panda (defaultPrim) (Xform)
    panda_link0 (Xform)
        panda_joint1 (PhysicsRevoluteJoint)
        visuals (Xform)
            material_Face636_001 (Material)
                Shader (Shader)
            material_Part__Feature017_001 (Material)
                Shader (Shader)
            visuals (Mesh)
                material_0 (GeomSubset)
                material_1 (GeomSubset)
        collisions (Xform)
            collisions (Mesh)
```

The structure is different, so I think the approach outlined in the [instanceable assets documentation](https://docs.omniverse.nvidia.com/isaacsim/latest/isaac_lab_tutorials/tutorial_instanceable_assets.html?highlight=instanceable#modifying-existing-assets) cannot be directly applied.

Are there any examples or best practices for making USDs like franka_alt_fingers.usd or other custom USDs instanceable?

Any help or suggestions would be appreciated!

**Comment by kellyguo11** [comment]
To make an existing USD instanceable, you can follow the guideline at https://docs.omniverse.nvidia.com/isaacsim/latest/isaac_lab_tutorials/tutorial_instanceable_assets.html?highlight=instanceable#hierarchy-requirement-for-instanceable-assets to understand the structure requirement for instanceable assets. The conversion process to bring an existing USD into the required structure could be a bit manual, since each asset is designed in different ways. The easiest method to make an asset instanceable is through the importers directly from MJCF or URDF. These importers will take care of the structure conversion to make sure it can work as an instanceable asset.

**Reply by right-chan** [reply]
Thank you. I'll have to configure it to use the importer.
